### PR TITLE
add $attributes to ResolvedReference

### LINKED ISSUE #14: Enhance ResolvedReference
Hello,

the class `Doctrine\RST\References\ResolvedReference` misses some way to add html attributes to the generated `<a>` 
Maybe we could add an `$htmlAttributes` associative array to this class, and then generate the expected markup in `\Doctrine\RST\Span::render()`?

thanks :)

### PR DESCRIPTION
**Pull request by nikophil**
fixes #14

### CODE REVIEW

File: lib/HTML/Span.php [hunk 1]
```diff
@@ -36,9 +37,17 @@ public function literal(string $text) : string
...
-    public function link(?string $url, string $title) : string
+    /**
+     * @param string[] $attributes
+     */
+    public function link(?string $url, string $title, array $attributes = []) : string
     {
-        return '<a href="' . htmlspecialchars((string) $url) . '">' . $title . '</a>';
+        $htmlAttributes = '';
+        foreach ($attributes as $attribute => $value) {
+            $htmlAttributes .= sprintf(' %s="%s"', $attribute, $value);
```
[the remark this answers is not in the document]
Comment (nikophil) [reply lib/HTML/Span.php]: do you want that i test if attribute names against a regex ? or maybe against a list of valid attributes ?

[the remark this answers is not in the document]
Comment (Ocramius) [reply lib/HTML/Span.php]: Any approach preventing invalid attribute names would be OK. An exception is ok if escaping is not possible.

File: lib/References/ResolvedReference.php [hunk 2]
```diff
@@ -42,4 +52,21 @@ public function getTitles() : array
...
+
+    /**
+     * @return string[]
+     */
+    public function getAttributes() : array
+    {
+        return $this->attributes;
+    }
+
+    protected function validateAttributes() : void
```
Comment (Ocramius) [inline lib/References/ResolvedReference.php]: `private`

Comment (Ocramius) [reply lib/References/ResolvedReference.php]: Make the attributes a parameter, avoid setting them to property until after calling this method

File: tests/References/ResolvedReferenceTest.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+        $this->assertSame('title', $resolvedReference->getTitle());
+        $this->assertSame('url', $resolvedReference->getUrl());
+        $this->assertSame(['title' => 'title'], $resolvedReference->getTitles());
+        $this->assertSame($attributes, $resolvedReference->getAttributes());
+    }
+
+    public function attributesValid() : \Generator
+    {
+        yield 'attributes #1' => [
```
Comment (Ocramius) [inline tests/References/ResolvedReferenceTest.php]: Note: the `yield` approach is only useful when deferred data provider entries are required: favour a simple `array` when not needed.

**Comment by nikophil** [comment]
hi @Ocramius 
i realized that we might also need to pass some attributes through the actual `Resolver`, then i made it and test the `Resolver`

**Pull request merged**